### PR TITLE
Allow minimum-size test to never skip

### PR DESCRIPTION
**Pull request by slinksoft**
The Capstone 1 team at FIU attempted to solve easy issue #434  for our Senior Project class. This may not be a probable solution, since it feels like the test results for the 2 skipped tests may be inaccurate; however, we tested it in our environment and it is working in some way.

image

### CODE REVIEW

File: tests/test_struct.py [hunk 1]
```diff
@@ -176,7 +176,9 @@ def _test_unpack(self, pyof_obj, bytes2unpack=None):
 
     def test_minimum_size(self):
         """Test struct minimum size."""
-        if self._min_size is None:
-            raise self.skipTest('minimum size was not set.')
+        """Temporary solution to solve issue of minimum size test skipping in
```
Comment (hdiogenes) [inline tests/test_struct.py:179]: The method ended up with two docstrings, please fix that. This branch also needs a rebase from the master branch.

Comment (slinksoft) [reply tests/test_struct.py:179]: The two docstrings is something that happened while we were learning Python at the beginning of the semester, so we apologize about that simple mistake. We will fix this as soon as we can.

**Review by gleybersonandrade — APPROVED** [review]

**Comment by hdiogenes** [comment]
@slinksoft please squash and fix your author name on the commits to match the CLA.

File: tests/unit/test_struct.py [hunk 2]
```diff
@@ -176,7 +176,9 @@ def _test_unpack(self, pyof_obj, bytes2unpack=None):
 
     def test_minimum_size(self):
         """Test struct minimum size."""
-        if self._min_size is None:
-            raise self.skipTest('minimum size was not set.')
+        # Temporary solution to solve issue of minimum size test skipping in
+	    # some test cases
```
Comment (hdiogenes) [inline tests/unit/test_struct.py:180]:
```suggestion
	    # some test cases
```
```suggestion
        # Temporary solution to solve issue of minimum size test skipping in
	  # some test cases
```

**Comment by hdiogenes** [comment]
@slinksoft, I've added a note to the original issue (#434). Following the suggestion there, you should force the test to error when the minimum size is not set:

```
diff --git a/tests/unit/test_struct.py b/tests/unit/test_struct.py
index 8316e9c..e6918b1 100644
--- a/tests/unit/test_struct.py
+++ b/tests/unit/test_struct.py
@@ -176,9 +176,7 @@ class TestStruct(unittest.TestCase):
 
     def test_minimum_size(self):
         """Test struct minimum size."""
-        # Temporary solution to solve issue of minimum size test skipping in
-           # some test cases
         obj = TestStruct._msg_cls()
         if self._min_size is None:
-            self._min_size = obj.get_size()
+            raise Exception(f'{self.__class__.__name__}._min_size is not set')
         self.assertEqual(obj.get_size(), self._min_size)

```

Then you will see which classes need to have the minimum size defined according to the spec. Luckily there are only two:

```
========================================================================= FAILURES ==========================================================================
_____________________________________________________________ TestFlowDelete.test_minimum_size ______________________________________________________________

self = <tests.unit.v0x01.test_controller2switch.test_flow_mod.TestFlowDelete testMethod=test_minimum_size>

    def test_minimum_size(self):
        """Test struct minimum size."""
        obj = TestStruct._msg_cls()
        if self._min_size is None:
>           raise Exception(f'{self.__class__.__name__}._min_size is not set')
E           Exception: TestFlowDelete._min_size is not set

tests/unit/test_struct.py:181: Exception
____________________________________________________________ TestVendorHeader.test_minimum_size _____________________________________________________________

self = <tests.unit.v0x01.test_symmetric.test_vendor_header.TestVendorHeader testMethod=test_minimum_size>

    def test_minimum_size(self):
        """Test struct minimum size."""
        obj = TestStruct._msg_cls()
        if self._min_size is None:
>           raise Exception(f'{self.__class__.__name__}._min_size is not set')
E           Exception: TestVendorHeader._min_size is not set

tests/unit/test_struct.py:181: Exception
================================================================== short test summary info ==================================================================
FAILED tests/unit/v0x01/test_controller2switch/test_flow_mod.py::TestFlowDelete::test_minimum_size - Exception: TestFlowDelete._min_size is not set
FAILED tests/unit/v0x01/test_symmetric/test_vendor_header.py::TestVendorHeader::test_minimum_size - Exception: TestVendorHeader._min_size is not set
=================================================== 2 failed, 392 passed, 46 skipped, 1 xfailed in 15.18s ===================================================

```

Then it's just a matter of defining `_min_size` on those classes.

**Comment by slinksoft** [comment]
> 
> 
> @slinksoft, I've added a note to the original issue (#434). Following the suggestion there, you should force the test to error when the minimum size is not set:
> 
> ```
> diff --git a/tests/unit/test_struct.py b/tests/unit/test_struct.py
> index 8316e9c..e6918b1 100644
> --- a/tests/unit/test_struct.py
> +++ b/tests/unit/test_struct.py
> @@ -176,9 +176,7 @@ class TestStruct(unittest.TestCase):
>  
>      def test_minimum_size(self):
>          """Test struct minimum size."""
> -        # Temporary solution to solve issue of minimum size test skipping in
> -           # some test cases
>          obj = TestStruct._msg_cls()
>          if self._min_size is None:
> -            self._min_size = obj.get_size()
> +            raise Exception(f'{self.__class__.__name__}._min_size is not set')
>          self.assertEqual(obj.get_size(), self._min_size)
> ```
> 
> Then you will see which classes need to have the minimum size defined according to the spec. Luckily there are only two:
> 
> ```
> ========================================================================= FAILURES ==========================================================================
> _____________________________________________________________ TestFlowDelete.test_minimum_size ______________________________________________________________
> 
> self = <tests.unit.v0x01.test_controller2switch.test_flow_mod.TestFlowDelete testMethod=test_minimum_size>
> 
>     def test_minimum_size(self):
>         """Test struct minimum size."""
>         obj = TestStruct._msg_cls()
>         if self._min_size is None:
> >           raise Exception(f'{self.__class__.__name__}._min_size is not set')
> E           Exception: TestFlowDelete._min_size is not set
> 
> tests/unit/test_struct.py:181: Exception
> ____________________________________________________________ TestVendorHeader.test_minimum_size _____________________________________________________________
> 
> self = <tests.unit.v0x01.test_symmetric.test_vendor_header.TestVendorHeader testMethod=test_minimum_size>
> 
>     def test_minimum_size(self):
>         """Test struct minimum size."""
>         obj = TestStruct._msg_cls()
>         if self._min_size is None:
> >           raise Exception(f'{self.__class__.__name__}._min_size is not set')
> E           Exception: TestVendorHeader._min_size is not set
> 
> tests/unit/test_struct.py:181: Exception
> ================================================================== short test summary info ==================================================================
> FAILED tests/unit/v0x01/test_controller2switch/test_flow_mod.py::TestFlowDelete::test_minimum_size - Exception: TestFlowDelete._min_size is not set
> FAILED tests/unit/v0x01/test_symmetric/test_vendor_header.py::TestVendorHeader::test_minimum_size - Exception: TestVendorHeader._min_size is not set
> =================================================== 2 failed, 392 passed, 46 skipped, 1 xfailed in 15.18s ===================================================
> ```
> 
> Then it's just a matter of defining `_min_size` on those classes.

Everything here is done. As mentioned in the kytos vlan_pool PR, I submitted a new CLA with the CLA matching BOTH my display name (my actual name) and my Github account/username. They SHOULD match now.

**Pull request merged**